archive/tar_stream.ts

```diff
@@ -0,0 +1,466 @@
+        const options: Required<TarStreamOptions> = {
+          mode: typeflag === "5" ? 755 : 644,
+          uid: 0,
+          gid: 0,
+          mtime: Math.floor(new Date().getTime() / 1000),
+          uname: "",
+          gname: "",
+          devmajor: "",
+          devminor: "",
+          ...chunk.options,
+        };
+
+        header.set(name); // name
+        header.set(
+          encoder.encode(
+            options.mode.toString().padStart(6, "0") + " \0" + // mode
+              options.uid.toString().padStart(6, "0") + " \0" + //uid
+              options.gid.toString().padStart(6, "0") + " \0" + // gid
+              size.toString(8).padStart(size < 8 ** 11 ? 11 : 12, "0") +
+              (size < 8 ** 11 ? " " : "") + // size
+              options.mtime.toString(8).padStart(11, "0") + " " + // mtime
+              " ".repeat(8) + // checksum | To be updated later
+              typeflag + // typeflag
+              "\0".repeat(100) + // linkname
+              "ustar\0" + // magic
+              "00" + // version
+              options.uname.padStart(32, "\0") + // uname
+              options.gname.padStart(32, "\0") + // gname
+              options.devmajor.padStart(8, "\0") + // devmajor
+              options.devminor.padStart(8, "\0"), // devminor
+          ),
+          100,
+        );
+        header.set(prefix, 345); // prefix
+        // Update Checksum
+        header.set(
+          encoder.encode(
+            header.reduce((x, y) => x + y).toString(8).padStart(6, "0") + "\0",
+          ),
+          148,
+        );
+        yield header;
+
+        if ("size" in chunk) {
+          let size = 0;
+          for await (const value of chunk.readable) {
+            size += value.length;
+            yield value;
+          }
+          if (chunk.size !== size) {
+            throw new RangeError(
+              "Provided size did not match bytes read from provided readable",
+            );
+          }
+          if (chunk.size % 512) {
+            yield new Uint8Array(512 - size % 512);
+          }
+        }
+      }
+      yield new Uint8Array(1024);
+    }();
+    this.#readable = new ReadableStream({
+      type: "bytes",
+      async pull(controller) {
+        const { done, value } = await gen.next();
+        if (done) {
+          controller.close();
+          return controller.byobRequest?.respond(0);
+        }
+        if (controller.byobRequest?.view) {
+          const buffer = new Uint8Array(controller.byobRequest.view.buffer);
+
+          const size = buffer.length;
+          if (size < value.length) {
+            buffer.set(value.slice(0, size));
+            controller.byobRequest.respond(size);
+            controller.enqueue(value.slice(size));
+          } else {
+            buffer.set(value);
+            controller.byobRequest.respond(value.length);
+          }
+        } else {
+          controller.enqueue(value);
+        }
+      },
+    });
```

Comment:
> The reason why this PR is wroking and mine is not is because this one does not use byte streams. Not sure if this should be a blocking action or not, but its definitively a downgrade when comparing to the non-stream API and doesnt line up with other streams we do in CLI that do file or io handling.
- https://github.com/denoland/std/pull/4548#pullrequestreview-1988145574

> Looking into what bytes streams are and how they work, I do think I could implement them into both streams. At least for when they're serving content out. I don't think I'd run into the same problem you're facing in implementing byte streams with it closing early or something. 
- https://github.com/denoland/std/pull/4548#issuecomment-2065889430

From my understanding the Deno team want TransformStreams to be byte streams where possible.

Reply:
Thoughts, @crowlKats?

Reply:
I havent followed this PR too closely, but ideally the streams resulted from this transform would be byte streams